### PR TITLE
fix hanging tests and allow for single threaded executor

### PR DESCRIPTION
**Pull request by dignifiedquire**

### CODE REVIEW

File: src/provider/mod.rs [hunk 1]
```diff
@@ -654,13 +661,13 @@ impl RpcHandler {
     async fn id(self, _: IdRequest) -> IdResponse {
         IdResponse {
             peer_id: Box::new(self.inner.keypair.public().into()),
-            listen_addrs: self.inner.listen_addresses().await.unwrap_or_default(),
+            listen_addrs: self.inner.listen_addresses().unwrap_or_default(),
             version: env!("CARGO_PKG_VERSION").to_string(),
         }
     }
     async fn addrs(self, _: AddrsRequest) -> AddrsResponse {
```
Comment (rklaehn) [inline src/provider/mod.rs:668]: this does not have to be async anymore, right?

File: src/provider/mod.rs [hunk 2]
```diff
@@ -654,13 +661,13 @@ impl RpcHandler {
     async fn id(self, _: IdRequest) -> IdResponse {
```
Comment (rklaehn) [inline src/provider/mod.rs:661]: Also not async anymore

**Review by rklaehn — APPROVED** [review]
So why was it failing before?

This mix of an actor approach and local caching is a bit weird, even though less async makes it more pleasant to use.

File: src/provider/mod.rs (hunk 1 above)
Comment (dignifiedquire) [reply src/provider/mod.rs:668]: yes, but the rpc code was upset when I made it not async 🤷

**Comment by dignifiedquire** [comment]
> This mix of an actor approach and local caching is a bit weird,

It is, would love to hear any suggestions on alternatives. This was the least bad one I could come up with.

**Comment by dignifiedquire** [comment]
armv7 should be fixed when we upgrade socket2: https://github.com/rust-lang/socket2/pull/434

**Review by flub — APPROVED** [review]

**Pull request merged**